src/main/scala/Table.scala

```diff
@@ -492,35 +508,71 @@ case class Table private (
     annotations: Map[String, JsonNode],
     var warnings: Array[ErrorWithCsvContext]
 ) {
-  var foreignKeyValues: Map[String, JsonNode] = Map()
 
   /**
     * This array contains the foreign keys defined in other tables' schemas which reference data inside this table.
     */
   var foreignKeyReferences: Array[ForeignKeyWithTable] = Array()
-  var foreignKeyReferenceValues: Map[String, JsonNode] = Map()
-  var warningsFromColumns: Array[ErrorWithCsvContext] =
-    Array[ErrorWithCsvContext]()
-  for (c <- columns) {
-    for (w <- c.warnings) {
-      warningsFromColumns :+= w
-    }
-  }
-  warnings = warnings.concat(warningsFromColumns)
 
-  def validateRow(row: CSVRecord): WarningsAndErrors = {
+  def validateRow(row: CSVRecord): Option[ValidateRowOutput] = {
     var errors = Array[ErrorWithCsvContext]()
+    var primaryKeyValues = List[Any]()
```

Comment:
The initialisation of these lists can happen inside your if statement.

Reply:
ok